Code/Framework/AzCore/AzCore/Memory/HphaAllocator.cpp

```diff
@@ -1556,6 +1536,116 @@ namespace AZ
         return nullptr;
     }
 
+    template<bool DebugAllocatorEnable>
+    auto HphaSchemaBase<DebugAllocatorEnable>::HpAllocator::tree_grow_aligned(size_t size, size_t alignment) -> block_header*
+    {
+        const size_t sizeWithBlockHeaders = size + 3 * sizeof(block_header); // two fences plus one fake
+
+        // tree_system_alloc() will return an offset that is a multiple of OS_VIRTUAL_PAGE_SIZE which is 64KiB
```

Comment:
I wonder if its better to put these longer explainations and details in a mark down file in this folder (something like HphaAllocator.md) and then just reference that file in the code instead.

Reply:
That could be arranged.